flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/linalg/BLAS.java

```diff
@@ -146,4 +148,38 @@ public static void gemv(double alpha, DenseMatrix matA, boolean transA,
 		final String ta = transA ? "T" : "N";
 		NATIVE_BLAS.dgemv(ta, m, n, alpha, matA.getData(), lda, x.getData(), 1, beta, y.getData(), 1);
 	}
+
+	/**
+	 * y := alpha * A * x + beta * y .
+	 */
+	public static void gemv(double alpha, DenseMatrix matA, boolean transA,
+							SparseVector x, double beta, DenseVector y) {
+		if (transA) {
```

Comment:
Can we create a `transposePreconditionChecker`? feel like this would be use multiple place/times with duplicate code in such as `gemv`, `gemm`, etc.

Reply:
Yes, you are right. We have created a method "gemvDimensionCheck". However, the dimension check in "gemm" and "gemv" are totally different, so "gemm" is left untouched.